test/integration/singlecluster/scheduler/resourcetransformations/resource_transformations_test.go

```diff
@@ -0,0 +1,192 @@
+		util.MustCreate(ctx, k8sClient, defaultFlavor)
+
+		ns = util.CreateNamespaceFromPrefixWithLog(ctx, k8sClient, "resource-transformations-")
+
+		clusterQueue = utiltestingapi.MakeClusterQueue("test-cq").
+			ResourceGroup(
+				*utiltestingapi.MakeFlavorQuotas("default").
+					Resource("example.com/accelerator-memory", "100Gi").
+					Resource("example.com/credits", "1000").
+					Resource("nvidia.com/gpu", "50").
+					Resource("nvidia.com/total-gpucores", "1000").
+					Resource("nvidia.com/total-gpumem", "102400").
+					Obj(),
+			).Obj()
+		util.MustCreate(ctx, k8sClient, clusterQueue)
+
+		localQueue = utiltestingapi.MakeLocalQueue("test-lq", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+		util.MustCreate(ctx, k8sClient, localQueue)
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		util.ExpectObjectToBeDeleted(ctx, k8sClient, clusterQueue, true)
+		util.ExpectObjectToBeDeleted(ctx, k8sClient, defaultFlavor, true)
+	})
+
+	ginkgo.AfterAll(func() {
+		fwk.StopManager(ctx)
+	})
+
+	ginkgo.It("should transform resources with MultiplyBy", func() {
+		ginkgo.By("Creating a workload with vgpu resources")
+		wl := utiltestingapi.MakeWorkload("multiply-wl", ns.Name).
+			Queue(kueue.LocalQueueName(localQueue.Name)).
+			Request("nvidia.com/gpu", "2").
+			Request("nvidia.com/gpucores", "20").
+			Request("nvidia.com/gpumem", "1024").
+			Obj()
+		util.MustCreate(ctx, k8sClient, wl)
+
+		ginkgo.By("Waiting for workload to be admitted")
+		util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, wl)
+
+		wlLookupKey := client.ObjectKeyFromObject(wl)
+		createdWorkload := &kueue.Workload{}
+
+		ginkgo.By("Verifying workload is created but not admitted yet", func() {
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, wlLookupKey, createdWorkload)).To(gomega.Succeed())
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		})
+
+		ginkgo.By("Verifying MultiplyBy transformation")
+		gomega.Expect(createdWorkload.Status.Admission).NotTo(gomega.BeNil())
+		resourceUsage := createdWorkload.Status.Admission.PodSetAssignments[0].ResourceUsage
+
+		// Original resources should be replaced
+		gomega.Expect(resourceUsage).NotTo(gomega.HaveKey(corev1.ResourceName("nvidia.com/gpucores")))
+		gomega.Expect(resourceUsage).NotTo(gomega.HaveKey(corev1.ResourceName("nvidia.com/gpumem")))
+
+		// Transformed resources: vgpucores * vgpu = 20 * 2 = 40
+		gomega.Expect(resourceUsage).To(gomega.HaveKey(corev1.ResourceName("nvidia.com/total-gpucores")))
+		gomega.Expect(resourceUsage["nvidia.com/total-gpucores"]).To(gomega.Equal(resource.MustParse("40")))
+
+		// Transformed resources: vgpumem * vgpu = 1024 * 2 = 2048
+		gomega.Expect(resourceUsage).To(gomega.HaveKey(corev1.ResourceName("nvidia.com/total-gpumem")))
+		gomega.Expect(resourceUsage["nvidia.com/total-gpumem"]).To(gomega.Equal(resource.MustParse("2048")))
```

Comment:
As commented in the other test I would prefer to assert on the entire structure: resourceUsage

Reply:
I agree with you.